src/operator/nn/dnnl/dnnl_concat-inl.h

```diff
@@ -52,13 +52,18 @@ class DNNLConcatFwd {
 
 static DNNLConcatFwd& GetConcatForward(int concat_dim,
                                        const std::vector<NDArray>& in_data,
-                                       const std::vector<dnnl::memory::desc>& data_md) {
+                                       const std::vector<dnnl::memory::desc>& data_md,
+                                       int cache_dim = -1) {
 #if DMLC_CXX11_THREAD_LOCAL
   static thread_local std::unordered_map<OpSignature, DNNLConcatFwd, OpHash> fwds;
 #else
   static MX_THREAD_LOCAL std::unordered_map<OpSignature, DNNLConcatFwd, OpHash> fwds;
 #endif
+  if (cache_dim == -1) {
+    cache_dim = concat_dim;
```

Comment:
Could you please tell me what is an advantage of adding the same value here (twofold: cache_dim if cache_dim == 1)?

Reply:
Changed to adding -1 value to key. It's necessary for stack op as there is used trick creating always 3 dimensional memory descriptor [leading_dim, mid_dim, trailing_dim]. For the same input shapes stack op can be performed on different axis. Without adding new key returned primitive would be the same for both scenarios which is wrong.